packages/era/src/exportHistory.ts

```diff
@@ -0,0 +1,158 @@
+  if (body === undefined) {
+    body = [[], [], []] as BlockBodyBytes
+  }
+  body = RLP.encode(body)
+  return { header, body }
+}
+
+export async function getBlockReceipts(DB: BlockDB, blockHash: Uint8Array): Promise<Uint8Array> {
+  const dbKey = concatBytes(intToBytes(DBKey.Receipts), blockHash)
+  const receipts = await DB.get(dbKey)
+  return (receipts ?? RLP.encode([])) as Uint8Array
+}
+
+export async function getBlockTuple(chainDB: BlockDB, metaDB: BlockDB, number: bigint) {
+  const blockHash = await numberToHash(chainDB, number)
+  const { header, body } = await getBlock(chainDB, number)
+  const totalDifficulty = await getTotalDifficulty(chainDB, blockHash, number)
+  const receipts = await getBlockReceipts(metaDB, blockHash)
+  return { blockHash, header, body, totalDifficulty, receipts }
+}
+
+export async function getBlocks(chainDB: BlockDB, metaDB: BlockDB, start: bigint, number: number) {
+  const blocks: {
+    blockHash: Uint8Array
+    header: Uint8Array
+    body: Uint8Array
+    totalDifficulty: bigint
+    receipts: Uint8Array
+  }[] = []
+  for (let i = 0; i < number; i++) {
+    const { blockHash, header, body, totalDifficulty, receipts } = await getBlockTuple(
+      chainDB,
+      metaDB,
+      start + BigInt(i),
+    )
+    blocks.push({ blockHash, header, body, totalDifficulty, receipts })
+  }
+  return blocks
+}
+
+async function getEpoch(chainDB: BlockDB, metaDB: BlockDB, index: number) {
+  const blocks = await getBlocks(chainDB, metaDB, BigInt(index * 8192), 8192)
+  const headerRecords = blocks.map((block) => {
+    return {
+      blockHash: block.blockHash,
+      totalDifficulty: block.totalDifficulty,
+    }
+  })
+  const blockTuples = blocks.map((block) => {
+    return {
+      header: block.header,
+      body: block.body,
+      receipts: block.receipts,
+      totalDifficulty: block.totalDifficulty,
+    }
+  })
+  return { blockTuples, headerRecords }
+}
+
+function initDBs(dataDir: string, chain: string) {
+  const chainDir = `${dataDir}/${chain}`
+  // Chain DB
+  const chainDataDir = `${chainDir}/chain`
+  const chainDB = new Level<string | Uint8Array, string | Uint8Array>(chainDataDir)
+
+  // Meta DB (receipts, logs, indexes, skeleton chain)
+  const metaDataDir = `${chainDir}/meta`
+  const metaDB = new Level<string | Uint8Array, string | Uint8Array>(metaDataDir)
+
+  return { chainDB, metaDB }
+}
+
+export async function exportEpochAsEra1(
+  epoch: number,
+  dataDir: string,
+  outputDir: string = dataDir,
+  chain = 'mainnet',
+) {
+  const { chainDB, metaDB } = initDBs(dataDir, chain)
+  await chainDB.open()
+  await metaDB.open()
+  const { blockTuples, headerRecords } = await getEpoch(chainDB, metaDB, epoch)
+  const era1 = await formatEra1(blockTuples, headerRecords, epoch)
+  writeFileSync(`${outputDir}/era1/epoch-${epoch}.era1`, era1)
+
+  // Create era1 directory if it doesn't exist
+  const era1Dir = `${outputDir}/era1`
```

Comment:
Does this work or did I put it in the wrong place?  This should go before you write the file or you'll get errors if the directory doesn't exist.